### PR TITLE
DS3: Add DLC, DLC Progressive Locations, and add Weapon Levels Options

### PR DESCRIPTION
**Pull request by Br00ty**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
Adds DLC, DLC Progressive locations, and add adjustable Weapon Level Options

## How was this tested?
Through a AP test and through friends.

### CODE REVIEW

File: worlds/dark_souls_3/__init__.py [hunk 1]
```diff
@@ -175,7 +178,7 @@ def create_regions(self):
             dreg_heap_region.exits.append(Entrance(self.player, "Goto Ringed City", dreg_heap_region))
             self.multiworld.get_entrance("Goto Ringed City", self.player).connect(ringed_city_region)
 
-    # For each region, add the associated locations retrieved from the corresponding location_table
+    # For each region, add the associated locations retrieved from the corresponding =
```
Comment (Marechal-L) [inline worlds/dark_souls_3/__init__.py]: Is it a mistake or you wanted to write something else ?

File: worlds/dark_souls_3/Options.py [hunk 2]
```diff
@@ -29,10 +29,56 @@ class NoEquipLoadOption(Toggle):
...
+    basic: Only weapons that can be upgraded to +10
+    epic: Only weapons that can be upgraded to +5"""
     display_name = "Randomize weapons level"
+    option_none = 0
+    option_all = 1
+    option_basic = 2
+    option_epic = 3
+
+
+class RandomizeWeaponsLevelPercentageOption(Range):
```
Comment (Marechal-L) [inline worlds/dark_souls_3/Options.py:45]: This option is missing a display_name.
We should add display_name = "Percentage of randomized weapons" for example

File: worlds/dark_souls_3/__init__.py (hunk 1 above)
Comment (Br00ty) [reply worlds/dark_souls_3/__init__.py]: this was a typo, changed it back

File: worlds/dark_souls_3/Options.py (hunk 2 above)
Comment (Br00ty) [reply worlds/dark_souls_3/Options.py:45]: you are right! added the display name that we needed

**Pull request merged**